### DISCUSSION TITLE
An error occured while saving changes

### DISCUSSION THREAD

**Discussion by Cnfnbcn** [X2CommunityCore/xcom2-launcher#341] [Help]
**AML Version:** 1.5.0

**Description**
On first launch I get this error:
_Unable to read mod directories from 'XComEngine.ini'. See file 'AML.log' for details_

Then this warning:
_Unable to detect mod directories. Please select them manually from the settings dialog._

Ok, I searched online and some threads suggested to manually set mod directory in settings. So did I. But when trying to restart AML, which is required for this change to take effect, I get this error:
_An error occured while saving changes.
The cloud file provider is not running._

This completely prevents me from using AML whatsoever.

![screenshot-87-30-04-23](https://user-images.githubusercontent.com/78432987/235353234-a1e9d84a-5c46-47ff-9c6f-75d1acc08f17.png)
![screenshot-88-30-04-23](https://user-images.githubusercontent.com/78432987/235353235-6961ce16-b9c7-456d-b37e-a5ddc49b6b70.png)
![screenshot-89-30-04-23](https://user-images.githubusercontent.com/78432987/235353236-92301180-bb0e-4947-85fa-d8fe193b7da5.png)

[AML.log](https://github.com/X2CommunityCore/xcom2-launcher/files/11361313/AML.log)

**Comment by RevZero** [comment]
The error "The cloud file provider is not running." suggest, that you are using something like onedrive or Dropbox and the respective service is not running.
I don't know why exactly this is an issue for AML and the file it's trying to access, but maybe this points you in the right direction.

**Comment by Cnfnbcn** [comment]
> The error "The cloud file provider is not running." suggest, that you are using something like onedrive or Dropbox and the respective service is not running. I don't know why exactly this is an issue for AML and the file it's trying to access, but maybe this points you in the right direction.

I mean, I never actively used any desktop cloud services and completely shut down one drive on windows installation. Might it be an issue with Windows 11, though?

**Comment by RevZero** [comment]
Try Google with that error message. There were two or three similar reports in the AML Discord and IIRC it was an OneDrive issue in all cases.

Moving this to discussions, it's not an AML issue.

**Reply by Cnfnbcn** [reply]
I never bother writing an issue if I didn't Google it first. I genuinely found zero topics that would be any similar. And again, I don't use OneDrive and I'm sure I have it disabled. And even if I did use it – the error occurs only with AML. So even if it's not AML fault, it is at least related to it.

**Comment by Dragon-32** [comment]
Where's AML installed, this entry imply it's in a OneDrive folder
Executable location: 'C:\Users\vasya\OneDrive\Ðàáî÷èé ñòîë\AML\XCOM2 Launcher.exe'

I've installed mine in a directory I created under C:\Programs, also running on Win11 with (AFAIK) OneDrive disabled and no MS account.

**Reply by Cnfnbcn** [reply]
It is just the desktop folder🤷‍♀️
I think I now understand what's the issue. Desktop folder is by default inside OneDrive folder. And since I have OneDrive completely shut down, it doesn't work.
PS: Yeah, I moved it outside desktop and it works.
